mscore/prefsdialog.cpp

```diff
@@ -250,7 +251,247 @@ PreferenceDialog::PreferenceDialog(QWidget* parent)
+                                                }),
+                  new CustomPreferenceItem(PREF_EXPORT_MUSICXML_EXPORTBREAKS, exportAllLayouts,
+                                          [this]() { // apply function
+                                                if (exportAllLayouts->isChecked())
+                                                      preferences.setCustomPreference<MusicxmlExportBreaks>(PREF_EXPORT_MUSICXML_EXPORTBREAKS, MusicxmlExportBreaks::ALL);
+                                                },
+                                          []()  { // update function
+                                                ;
+                                                }),
+                  new CustomPreferenceItem(PREF_EXPORT_MUSICXML_EXPORTBREAKS, exportAllBreaks,
+                                          [this]() { // apply function
+                                                if (exportAllBreaks->isChecked())
+                                                      preferences.setCustomPreference<MusicxmlExportBreaks>(PREF_EXPORT_MUSICXML_EXPORTBREAKS, MusicxmlExportBreaks::ALL);
+                                                },
+                                          [this]() { // update function
+                                                exportAllBreaks->setChecked(preferences.musicxmlExportBreaks() == MusicxmlExportBreaks::ALL);
+                                                }),
+                  new CustomPreferenceItem(PREF_EXPORT_MUSICXML_EXPORTBREAKS, exportManualBreaks,
+                                          [this]() { // apply function
+                                                if (exportManualBreaks->isChecked())
+                                                      preferences.setCustomPreference<MusicxmlExportBreaks>(PREF_EXPORT_MUSICXML_EXPORTBREAKS, MusicxmlExportBreaks::MANUAL);
+                                                },
+                                          [this]() { // update function
+                                                exportManualBreaks->setChecked(preferences.musicxmlExportBreaks() == MusicxmlExportBreaks::MANUAL);
+                                                }),
+                  new CustomPreferenceItem(PREF_EXPORT_MUSICXML_EXPORTBREAKS, exportNoBreaks,
+                                          [this]() { // apply function
+                                                if (exportNoBreaks->isChecked())
+                                                      preferences.setCustomPreference<MusicxmlExportBreaks>(PREF_EXPORT_MUSICXML_EXPORTBREAKS, MusicxmlExportBreaks::NO);
+                                                },
+                                          [this]() { // update function
+                                                exportNoBreaks->setChecked(preferences.musicxmlExportBreaks() == MusicxmlExportBreaks::NO);
+                                                }),
+
+      };
+
```

Comment:
To be more specific, I would add the following line right here:
`static constexpr auto doNothing = []() { };`

And replace the macro right below

Reply:
1. It can't be `constexpr` because that isn't supported in C++14. The build will fail.
2. I wouldn't make it `static` as I don't see the need, but I don't care all that much either way.
3. I would define it at the top of the function, since it's conceivable that some of the earlier widgets might want to make use of it in the future. Sticking it right in the middle of things (even though that's right before the current first use) seems a little strange.

Reply:
You are right `constexpr` can't be used with lambdas until c++17. I didn't know that.

You clearly have more experience than me here, since I'm just a music student with a c++ hobby, so please correct me if I am wrong: I got the idea that `static` should be used whenever, just like `const

Reply:
The meaning of `static` depends on the context. In this context, it means that the variable should be created the first time the function is called and should be kept in memory for the rest of the duration of the program. (Without the `static` specifier, the variable would be created every time the function is called and destroyed every time the function returns.)

For a tiny variable like this it doesn't really make a difference, but I prefer not to make things `static` unless there's a specific reason to; for example, if it's a large object that's expensive to create/destroy, or if it needs to preserve state across calls.

Reply:
I agree that making it `static` would be slightly confusing with no real benefits.